pyaedt/modules/solutions.py

```diff
@@ -1438,586 +1354,467 @@ def assign_weight(self, a, b, taper="flat", port_cont=0):
-        )
 
+        if show:
+            show_legend = True
+            if len(curves) > 15:
+                show_legend = False
+            if is_polar:
+                return plot_polar_chart(
+                    curves,
+                    xlabel=x_key,
+                    ylabel=farfield_quantity,
+                    title=title,
+                    snapshot_path=export_image_path,
+                    show_legend=show_legend,
+                )
+            else:
+                return plot_2d_chart(
+                    curves,
+                    xlabel=x_key,
+                    ylabel=farfield_quantity,
+                    title=title,
+                    snapshot_path=export_image_path,
+                    show_legend=show_legend,
+                )
+        else:
+            return curves
+
+    # fmt: off
     @pyaedt_function_handler()
     def polar_plot_3d(
-        self,
-        qty_str="RealizedGain",
-        phi_scan=0,
-        theta_scan=0,
-        title="3D Plot",
-        convert_to_db=True,
-        export_image_path=None,
+            self,
+            farfield_quantity="RealizedGain",
+            phi_scan=0,
+            theta_scan=0,
+            title="3D Plot",
+            format_quantity="dB10",
+            export_image_path=None,
+            show=True,
+            **kwargs
     ):
-        """Create a 3d plot of specified quantity.
+        # fmt: on
+        """Create a 3D plot of a specified quantity.
 
         Parameters
         ----------
-        qty_str : str, optional
-            Quantity to plot. Default `"RealizedGain"`.
+        farfield_quantity : str, optional
+            Far field quantity to plot. The default is ``"RealizedGain"``.
+            Available functions are: ``"RealizedGain"``, ``"RealizedGain_Phi"``, ``"RealizedGain_Theta"``,
+            ``"rEPhi"``, ``"rETheta"``, and ``"rETotal"``.
         phi_scan : float, int, optional
-            Phi Scan Angle in degree. Default `0`.
+            Phi scan angle in degree. The default is ``0``.
         theta_scan : float, int, optional
-            Theta Scan Angle in degree. Default `0`.
+            Theta scan angle in degree. The default is ``0``.
         title : str, optional
-            Plot title. Default `"3D Plot"`.
-        convert_to_db : bool, optional
-            Either if the quantity has to be converted in db or not. Default is `True`.
+            Plot title. The default is ``"3D Plot"``.
+        format_quantity : str, optional
+            Conversion data function.
+            Available functions are: ``"abs"``, ``"ang"``, ``"dB10"``, ``"dB20"``, ``"deg"``, ``"imag"``, ``"norm"``,
+            and ``"real"``.
         export_image_path : str, optional
-            Full path to image file. Default is None to not export.
+            Full path for the image file. The default is ``None``, in which case a file is not exported.
+        show : bool, optional
```

Comment:
This one has a different "Returns" section. That section also seems to contradict what is shown here? Maybe it's just me not understanding?